Resources/views/Form/form_admin_fields.html.twig

```diff
@@ -166,7 +166,14 @@ file that was distributed with this source code.
         {% endif %}
         <label{% for attrname, attrvalue in label_attr %} {{ attrname }}="{{ attrvalue }}"{% endfor %}>
             {{- widget|raw -}}
-            {{- label is not same as(false) ? label|trans({}, translation_domain) -}}
+            {# if label is defined and not on embedded collection #}
+            {%- if label is not same as(false) -%}
+                {%- if name == '_delete' -%}
+                    <span class="sr-only">{{ label|trans({}, translation_domain) }}</span>
+                {%- else -%}
+                    {{ label|trans({}, translation_domain) }}
+                {%- endif -%}
+            {%- endif -%}
```

Comment:
Ok it turns out to be possible:
```php
public function isEmbeddedInCollectionTable(FormView $form)
{
    while ($form = $form->parent) {
        if (empty($form->vars['sonata_admin']) || empty($form->vars['sonata_admin']['field_description'])) {
            break;
        }
        /** @var FieldDescriptionInterface $fd */
        $fd = $form->vars['sonata_admin']['field_description'];
        if ($fd->getType() === 'sonata_type_collection' && $fd->getOption('inline') === 'table') {
            return true;
        }
    }
    return false;
}
```

Of course this should apply only for single-checkbox field labels.